### PR TITLE
Direct user to log in when attempting to use fileshare

### PR DESCRIPTION
**Pull request by bartoszWojciechO**

### CODE REVIEW

File: cli/messages.go [hunk 1]
```diff
@@ -278,6 +278,7 @@ Learn more:
 	MsgFileshareDisconnectedPeer          = "Peer is disconnected."
 	MsgFileshareFileNotFound              = "File not found."
 	MsgFileshareSocketNotFound            = "Enable Meshnet to share files. If Meshnet is already enabled, try disabling and enabling it again. Use \"nordvpn set meshnet on\" to enable it."
+	MsgFileshareUserNotLoggedIn           = "You are not logged in. Please log in and enable meshnet to share files."
```
Comment (bartoszWojciechO) [inline cli/messages.go:281]: Message will be updated with input from copywriting team.

File: cli/cli_fileshare.go [hunk 2]
```diff
@@ -109,7 +110,12 @@ func statusLoop(fileshareClient pb.FileshareClient, client transferStatusClient,
 // IsFileshareDaemonReachable returns error if fileshare daemon is not reachable, daemon not running
 // being the most likely cause
 func (c *cmd) IsFileshareDaemonReachable(ctx *cli.Context) error {
-	_, err := c.fileshareClient.Ping(context.Background(), &pb.Empty{})
+	resp, err := c.client.IsLoggedIn(context.Background(), &dpb.Empty{})
+	if err != nil || !resp.GetValue() {
+		return formatError(fmt.Errorf(MsgFileshareUserNotLoggedIn))
```
Comment (mariusSincovici) [inline cli/cli_fileshare.go:115]: [minor] isn't good the error `internal.ErrNotLoggedIn`? If not, maybe change also in `fileshareServiceErrorCodeToError` to return the same error message?

**Review by mariusSincovici — APPROVED** [review]

File: cli/cli_fileshare.go (hunk 2 above)
Comment (bartoszWojciechO) [reply cli/cli_fileshare.go:115]: `ServiceErrorCode_NOT_LOGGED_IN` is not returned anywhere anyway, so we might just delete this branch from `fileshareServiceErrorCodeToError`. It was probably used back when fileshare was just a part of main daemon.

File: cli/cli_fileshare.go [hunk 3]
```diff
@@ -109,7 +110,12 @@ func statusLoop(fileshareClient pb.FileshareClient, client transferStatusClient,
 // IsFileshareDaemonReachable returns error if fileshare daemon is not reachable, daemon not running
 // being the most likely cause
 func (c *cmd) IsFileshareDaemonReachable(ctx *cli.Context) error {
-	_, err := c.fileshareClient.Ping(context.Background(), &pb.Empty{})
+	resp, err := c.client.IsLoggedIn(context.Background(), &dpb.Empty{})
+	if err != nil || !resp.GetValue() {
```
Comment (keliramu) [inline cli/cli_fileshare.go:114]: Why `OR !resp.GetValue()` part condition is used, in what situation it can happen?

**Review by keliramu — APPROVED** [review]

File: cli/cli_fileshare.go (hunk 3 above)
Comment (bartoszWojciechO) [reply cli/cli_fileshare.go:114]: Actually the response is a `*pb.Bool`, so `GetValue` returns a bool where true = user is logged in, false = user is not logged in.

**Pull request merged**